diff -r 3c7e7ad07c8d server/ModelLoader/BodyInfo_impl.cpp
--- a/server/ModelLoader/BodyInfo_impl.cpp	Fri May 20 16:41:05 2011 +0900
+++ b/server/ModelLoader/BodyInfo_impl.cpp	Fri Jun 03 14:56:48 2011 +0900
@@ -224,7 +224,7 @@
             traverseShapeNodes(segmentNodes[i].get(), T, links_[index].shapeIndices, links_[index].inlinedShapeTransformMatrices, &topUrl());
             long e =links_[index].shapeIndices.length();
             segmentInfo->shapeIndices.length(e-s);
-            for(int j=0, k=s; i<e; i++)
+            for(int j=0, k=s; k<e; k++)
                 segmentInfo->shapeIndices[j++] = k;
             links_[index].segments[i] = segmentInfo;
         }
