diff -r cf3fd4609ad5 server/ModelLoader/ColladaWriter.h
--- a/server/ModelLoader/ColladaWriter.h	Fri Jun 10 17:55:38 2011 +0900
+++ b/server/ModelLoader/ColladaWriter.h	Thu Jun 23 17:32:37 2011 +0900
@@ -771,7 +771,7 @@
             string rigidsid = str(boost::format("rigid%d")%ilink);
             pmout->vrigidbodysids.push_back(rigidsid);
             rigid_body->setSid(rigidsid.c_str());
-            rigid_body->setName(link.name);
+            rigid_body->setName(link.segments[0].name);
             domRigid_body::domTechnique_commonRef ptec = daeSafeCast<domRigid_body::domTechnique_common>(rigid_body->add(COLLADA_ELEMENT_TECHNIQUE_COMMON));
             domTargetable_floatRef mass = daeSafeCast<domTargetable_float>(ptec->add(COLLADA_ELEMENT_MASS));
             mass->setValue(link.mass);
@@ -784,6 +784,8 @@
             evec.resize(3,3);
             eval.resize(3);
             hrp::calcEigenVectors(inertia,evec,eval);
+	    if (det(evec) < 0.0) /* fix for right-handed coordinates */
+	      evec(0,2) *= -1.0; evec(1,2) *= -1.0; evec(2,2) *= -1.0;
             DblArray12 tinertiaframe;
             for(int j = 0; j < 3; ++j) {
                 tinertiaframe[4*0+j] = evec(0,j);
@@ -1022,7 +1024,7 @@
         LINKOUTPUT out;
         string linksid = _GetLinkSid(ilink);
         domLinkRef pdomlink = daeSafeCast<domLink>(pkinparent->add(COLLADA_ELEMENT_LINK));
-        pdomlink->setName(plink.name);
+        pdomlink->setName(plink.segments[0].name);
         pdomlink->setSid(linksid.c_str());
 
         domNodeRef pnode = daeSafeCast<domNode>(pnodeparent->add(COLLADA_ELEMENT_NODE));
@@ -1030,7 +1032,7 @@
         pnode->setId( nodeid.c_str() );
         string nodesid = str(boost::format("node%d")%ilink);
         pnode->setSid(nodesid.c_str());
-        pnode->setName(plink.name);
+        pnode->setName(plink.segments[0].name);
 
         for(int igeom = 0; igeom < plink.shapeIndices.length(); ++igeom) {
             string geomid = _GetGeometryId(bodyInfo, ilink,igeom);
