--- server/ModelLoader/ColladaWriter.h.org	2012-07-15 16:52:34.072509751 +0900
+++ server/ModelLoader/ColladaWriter.h	2012-07-15 16:52:50.268509182 +0900
@@ -975,7 +975,7 @@
 			    if ( appearanceInfo.normalPerVertex == 1 ) {
 				pindices->getValue()[i++] = appearanceInfo.normalIndices[ind];
 			    } else {
-				pindices->getValue()[i++] = appearanceInfo.normalIndices[triangles[ind]];
+				pindices->getValue()[i++] = appearanceInfo.normalIndices[triangles[ind]/3];
 			    }
 			}
 			if (textures.length() > 0 && appearanceInfo.textureCoordIndices.length() > 0 ){
