diff -r cf3fd4609ad5 hrplib/hrpModel/Link.h
--- a/hrplib/hrpModel/Link.h	Fri Jun 10 17:55:38 2011 +0900
+++ b/hrplib/hrpModel/Link.h	Sun Jun 19 23:00:56 2011 +0900
@@ -78,6 +78,8 @@
         int jointId;  ///< jointId value written in a model file
 		
         std::string name;
+        std::string link_name;
+        std::string joint_name;
 
         enum JointType {
             FIXED_JOINT,   ///< fixed joint(0 dof)
diff -r cf3fd4609ad5 hrplib/hrpModel/ModelLoaderUtil.cpp
--- a/hrplib/hrpModel/ModelLoaderUtil.cpp	Fri Jun 10 17:55:38 2011 +0900
+++ b/hrplib/hrpModel/ModelLoaderUtil.cpp	Sun Jun 19 23:00:56 2011 +0900
@@ -201,6 +201,8 @@
         
     CORBA::String_var name0 = linkInfo.name;
     link->name = string( name0 );
+    link->joint_name = string( name0 );
+    link->link_name = linkInfo.segments[0].name;
     link->jointId = jointId;
         
     Vector3 relPos(linkInfo.translation[0], linkInfo.translation[1], linkInfo.translation[2]);
