diff -r 3c7e7ad07c8d GrxUIonEclipse-project-0.9.8/src/com/generalrobotix/ui/GrxPluginManager.java
--- a/GrxUIonEclipse-project-0.9.8/src/com/generalrobotix/ui/GrxPluginManager.java	Fri May 20 16:41:05 2011 +0900
+++ b/GrxUIonEclipse-project-0.9.8/src/com/generalrobotix/ui/GrxPluginManager.java	Thu Jun 09 21:21:19 2011 +0900
@@ -412,8 +412,18 @@
                 pi.title = (String) GrxBasePlugin.getField(cls, "TITLE", cls.getSimpleName()); //$NON-NLS-1$
                 pi.lastDir = new File(homePath_ + (String) GrxBasePlugin.getField(cls, "DEFAULT_DIR", "")); //$NON-NLS-1$ //$NON-NLS-2$
                 String ext = (String) GrxBasePlugin.getField(cls, "FILE_EXTENSION", null); //$NON-NLS-1$
-                if (ext != null)
-                    pi.filter = "*." + ext;// GrxGuiUtil.createFileFilter(ext); //$NON-NLS-1$
+                if (ext != null) {
+		    pi.filter = new String[1];
+		    pi.filter[0] = "*." + ext;// GrxGuiUtil.createFileFilter(ext); //$NON-NLS-1$
+		} else {
+		    String[] exts = (String[]) GrxBasePlugin.getField(cls, "FILE_EXTENSIONS", null); //$NON-NLS-1$
+		    if (exts != null) {
+			pi.filter = new String[exts.length];
+			for (int i=0;i<exts.length;i++) {
+			    pi.filter[i] = "*." + exts[i];// GrxGuiUtil.createFileFilter(ext); //$NON-NLS-1$
+			}
+		    }
+		}
                 pinfoMap_.put(cls, pi);
             }
 
@@ -479,9 +489,8 @@
         if (!f.isFile()){
         	FileDialog fdlg = new FileDialog(GrxUIPerspectiveFactory.getCurrentShell(), SWT.OPEN);
         	PluginInfo pi = pinfoMap_.get(cls);
-            String[] fe = { pi.filter };
             fdlg.setText(_url+"  "+MessageBundle.get("GrxPluginManager.fileDialog.notFound"));
-            fdlg.setFilterExtensions(fe);
+            fdlg.setFilterExtensions(pi.filter);
             fdlg.setFilterPath(pi.lastDir.getAbsolutePath());
             String fPath = fdlg.open();
             if (fPath != null) {
@@ -943,7 +952,7 @@
         static final String VISIBLE = "visible"; //$NON-NLS-1$
         String title;
         File lastDir;
-        String filter;
+        String[] filter;
         Vector<Action> menu;
         boolean visible;
     }
@@ -984,8 +993,7 @@
 
             public void run() {
                 FileDialog fdlg = new FileDialog(GrxUIPerspectiveFactory.getCurrentShell(), SWT.OPEN);
-                String[] fe = { pi.filter };
-                fdlg.setFilterExtensions(fe);
+                fdlg.setFilterExtensions(pi.filter);
                 fdlg.setFilterPath(pi.lastDir.getAbsolutePath());
                 String fPath = fdlg.open();
                 if (fPath != null) {
diff -r 3c7e7ad07c8d GrxUIonEclipse-project-0.9.8/src/com/generalrobotix/ui/item/GrxModelItem.java
--- a/GrxUIonEclipse-project-0.9.8/src/com/generalrobotix/ui/item/GrxModelItem.java	Fri May 20 16:41:05 2011 +0900
+++ b/GrxUIonEclipse-project-0.9.8/src/com/generalrobotix/ui/item/GrxModelItem.java	Thu Jun 09 21:21:19 2011 +0900
@@ -51,7 +51,7 @@
 public class GrxModelItem extends GrxBaseItem implements Manipulatable {
     public static final String TITLE = "Model"; //$NON-NLS-1$
     public static final String DEFAULT_DIR = "/../model"; //$NON-NLS-1$
-    public static final String FILE_EXTENSION = "wrl"; //$NON-NLS-1$
+    public static final String[] FILE_EXTENSIONS = {"wrl","dae","zae"}; //$NON-NLS-1$
     private static final double DEFAULT_RADIUS = 0.05;
 
     // icons
