*** build/hrpsys-base/idl/SDOPackage.idl.org	2011-06-13 02:01:21.817850200 +0900
--- build/hrpsys-base/idl/SDOPackage.idl	2011-06-13 02:01:35.607470777 +0900
***************
*** 37,43 ****
  #include <CosNotifyComm.idl>
  #endif
  #ifndef ORBIT2_IDL
! #define TypeCode CORBA::TypeCode
  #endif
  /** CORBA specific model for SDOs */
  
--- 37,43 ----
  #include <CosNotifyComm.idl>
  #endif
  #ifndef ORBIT2_IDL
! //#define TypeCode CORBA::TypeCode
  #endif
  /** CORBA specific model for SDOs */
  
***************
*** 120,126 ****
    struct Parameter
    {
      string         name;
!     TypeCode       type;
      AllowedValues  allowed_values;
    };
  
--- 120,126 ----
    struct Parameter
    {
      string         name;
!     CORBA::TypeCode       type;
      AllowedValues  allowed_values;
    };
  
