diff -r 3c7e7ad07c8d GrxUIonEclipse-project-0.9.8/src/com/generalrobotix/ui/view/Grx3DView.java
--- GrxUIonEclipse-project-0.9.8/src/com/generalrobotix/ui/view/Grx3DView.java	Fri May 20 16:41:05 2011 +0900
+++ GrxUIonEclipse-project-0.9.8/src/com/generalrobotix/ui/view/Grx3DView.java	Fri Jun 03 14:56:48 2011 +0900
@@ -1202,7 +1202,7 @@
 					int startPosition =0;
 					int endPosition = currentWorld_.getLogSize();  
 					double playRateLogTime_ = currentWorld_.getTime(startPosition);					
-					for (int position=startPosition; position < endPosition; position++) {
+					for (int position=startPosition; position <= endPosition; position++) {
 						if(!btnRec_.isSelected())break;
 						double time = currentWorld_.getTime(position);
 						if (time >= playRateLogTime_) {
diff -r 3c7e7ad07c8d server/ModelLoader/BodyInfo_impl.cpp
--- server/ModelLoader/BodyInfo_impl.cpp	Fri May 20 16:41:05 2011 +0900
+++ server/ModelLoader/BodyInfo_impl.cpp	Fri Jun 03 14:56:48 2011 +0900
@@ -224,7 +224,7 @@
             traverseShapeNodes(segmentNodes[i].get(), T, links_[index].shapeIndices, links_[index].inlinedShapeTransformMatrices, &topUrl());
             long e =links_[index].shapeIndices.length();
             segmentInfo->shapeIndices.length(e-s);
-            for(int j=0, k=s; i<e; i++)
+            for(int j=0, k=s; k<e; k++)
                 segmentInfo->shapeIndices[j++] = k;
             links_[index].segments[i] = segmentInfo;
         }
