diff -r 3c7e7ad07c8d GrxUIonEclipse-project-0.9.8/src/com/generalrobotix/ui/view/Grx3DView.java
--- GrxUIonEclipse-project-0.9.8/src/com/generalrobotix/ui/view/Grx3DView.java	Fri May 20 16:41:05 2011 +0900
+++ GrxUIonEclipse-project-0.9.8/src/com/generalrobotix/ui/view/Grx3DView.java	Fri Jun 03 14:56:48 2011 +0900
@@ -1202,7 +1202,7 @@
 					int startPosition =0;
 					int endPosition = currentWorld_.getLogSize();  
 					double playRateLogTime_ = currentWorld_.getTime(startPosition);					
-					for (int position=startPosition; position < endPosition; position++) {
+					for (int position=startPosition; position <= endPosition; position++) {
 						if(!btnRec_.isSelected())break;
 						double time = currentWorld_.getTime(position);
 						if (time >= playRateLogTime_) {
diff -r 3c7e7ad07c8d server/ModelLoader/BodyInfo_impl.cpp
--- server/ModelLoader/BodyInfo_impl.cpp	Fri May 20 16:41:05 2011 +0900
+++ server/ModelLoader/BodyInfo_impl.cpp	Fri Jun 03 14:56:48 2011 +0900
@@ -224,7 +224,7 @@
             traverseShapeNodes(segmentNodes[i].get(), T, links_[index].shapeIndices, links_[index].inlinedShapeTransformMatrices, &topUrl());
             long e =links_[index].shapeIndices.length();
             segmentInfo->shapeIndices.length(e-s);
-            for(int j=0, k=s; i<e; i++)
+            for(int j=0, k=s; k<e; k++)
                 segmentInfo->shapeIndices[j++] = k;
             links_[index].segments[i] = segmentInfo;
         }
diff -r 3c7e7ad07c8d server/ControllerBridge/main.cpp
--- server/ControllerBridge/main.cpp	Fri May 20 16:41:05 2011 +0900
+++ server/ControllerBridge/main.cpp	Fri Jun 03 20:49:26 2011 +0900
@@ -97,16 +97,18 @@
 
     try {
 		unsigned int i;
+		int rtc_argc = 1;
+		char** rtc_argv = (char **)malloc(sizeof(char *)*argc);
+		rtc_argv[0] = argv[0];
 		for (i=1; i<argc; i++){
-			if (strcmp(argv[i], "-f")==0){
-				break;
-			} 
+			if (strncmp(argv[i], "--", 2)!=0 ) {
+				rtc_argv[rtc_argc] = argv[i];
+				rtc_argc++;
+			}  else {
+				i++;
+			}
 		}
-		if (i == argc){
-			rtcManager = RTC::Manager::init(1, argv);
-		}else{
-			rtcManager = RTC::Manager::init(3, argv+i-1);
-		}
+		rtcManager = RTC::Manager::init(rtc_argc, rtc_argv);
 		rtcManager->activateManager();
 	}
 	catch(...) {
diff -r 3c7e7ad07c8d GrxUIonEclipse-project-0.9.8/src/com/generalrobotix/ui/GrxPluginManager.java
--- GrxUIonEclipse-project-0.9.8/src/com/generalrobotix/ui/GrxPluginManager.java	Fri May 20 16:41:05 2011 +0900
+++ GrxUIonEclipse-project-0.9.8/src/com/generalrobotix/ui/GrxPluginManager.java	Thu Jun 09 21:21:19 2011 +0900
@@ -412,8 +412,18 @@
                 pi.title = (String) GrxBasePlugin.getField(cls, "TITLE", cls.getSimpleName()); //$NON-NLS-1$
                 pi.lastDir = new File(homePath_ + (String) GrxBasePlugin.getField(cls, "DEFAULT_DIR", "")); //$NON-NLS-1$ //$NON-NLS-2$
                 String ext = (String) GrxBasePlugin.getField(cls, "FILE_EXTENSION", null); //$NON-NLS-1$
-                if (ext != null)
-                    pi.filter = "*." + ext;// GrxGuiUtil.createFileFilter(ext); //$NON-NLS-1$
+                if (ext != null) {
+		    pi.filter = new String[1];
+		    pi.filter[0] = "*." + ext;// GrxGuiUtil.createFileFilter(ext); //$NON-NLS-1$
+		} else {
+		    String[] exts = (String[]) GrxBasePlugin.getField(cls, "FILE_EXTENSIONS", null); //$NON-NLS-1$
+		    if (exts != null) {
+			pi.filter = new String[exts.length];
+			for (int i=0;i<exts.length;i++) {
+			    pi.filter[i] = "*." + exts[i];// GrxGuiUtil.createFileFilter(ext); //$NON-NLS-1$
+			}
+		    }
+		}
                 pinfoMap_.put(cls, pi);
             }
 
@@ -479,9 +489,8 @@
         if (!f.isFile()){
         	FileDialog fdlg = new FileDialog(GrxUIPerspectiveFactory.getCurrentShell(), SWT.OPEN);
         	PluginInfo pi = pinfoMap_.get(cls);
-            String[] fe = { pi.filter };
             fdlg.setText(_url+"  "+MessageBundle.get("GrxPluginManager.fileDialog.notFound"));
-            fdlg.setFilterExtensions(fe);
+            fdlg.setFilterExtensions(pi.filter);
             fdlg.setFilterPath(pi.lastDir.getAbsolutePath());
             String fPath = fdlg.open();
             if (fPath != null) {
@@ -943,7 +952,7 @@
         static final String VISIBLE = "visible"; //$NON-NLS-1$
         String title;
         File lastDir;
-        String filter;
+        String[] filter;
         Vector<Action> menu;
         boolean visible;
     }
@@ -984,8 +993,7 @@
 
             public void run() {
                 FileDialog fdlg = new FileDialog(GrxUIPerspectiveFactory.getCurrentShell(), SWT.OPEN);
-                String[] fe = { pi.filter };
-                fdlg.setFilterExtensions(fe);
+                fdlg.setFilterExtensions(pi.filter);
                 fdlg.setFilterPath(pi.lastDir.getAbsolutePath());
                 String fPath = fdlg.open();
                 if (fPath != null) {
diff -r 3c7e7ad07c8d GrxUIonEclipse-project-0.9.8/src/com/generalrobotix/ui/item/GrxModelItem.java
--- GrxUIonEclipse-project-0.9.8/src/com/generalrobotix/ui/item/GrxModelItem.java	Fri May 20 16:41:05 2011 +0900
+++ GrxUIonEclipse-project-0.9.8/src/com/generalrobotix/ui/item/GrxModelItem.java	Thu Jun 09 21:21:19 2011 +0900
@@ -51,7 +51,7 @@
 public class GrxModelItem extends GrxBaseItem implements Manipulatable {
     public static final String TITLE = "Model"; //$NON-NLS-1$
     public static final String DEFAULT_DIR = "/../model"; //$NON-NLS-1$
-    public static final String FILE_EXTENSION = "wrl"; //$NON-NLS-1$
+    public static final String[] FILE_EXTENSIONS = {"wrl","dae","zae"}; //$NON-NLS-1$
     private static final double DEFAULT_RADIUS = 0.05;
 
     // icons
