--- src/Base/OptionManager.cpp.org	2012-01-06 11:21:09.398429621 +0900
+++ src/Base/OptionManager.cpp	2012-01-06 11:20:06.494428624 +0900
@@ -31,6 +31,8 @@
     }
     
     info->options.add_options()("help,h", "show help message");
+    info->options.add_options()("dummy", program_options::value<vector<string> >());
+    info->positionalOptions.add("dummy", -1);
 
     bool is_error = false;
     try{
