Index: Grasp/CMakeLists.txt
===================================================================
--- Grasp/CMakeLists.txt	(リビジョン 10)
+++ Grasp/CMakeLists.txt	(作業コピー)
@@ -43,7 +43,7 @@
 
 if(UNIX)
 
-	target_link_libraries(${libname} CnoidBodyPlugin qhull ${PYTHON_LIBRARIES} ${Boost_PYTHON_LIBRARY})
+	target_link_libraries(${libname} CnoidBodyPlugin qhull ${PYTHON_LIBRARIES} ${Boost_PYTHON_LIBRARY} boost_regex)
 
 
   install(TARGETS ${libname}
