--- rtshell/fmt.py.org	2012-01-06 18:37:15.814844054 +0900
+++ rtshell/fmt.py	2012-01-06 18:38:37.194845343 +0900
@@ -18,6 +18,9 @@
 
 '''
 
+import inspect
+import rts_exceptions
+
 
 ###############################################################################
 ## Python source formatter
@@ -41,6 +44,7 @@
 
     '''
     # Special case for internal formatters: replace 'rtshell' with 'fmt'
+    form_rpl = form
     if form.startswith('rtshell.'):
         form_rpl = 'fmt.' + form[8:]
     try:
