+++ RTC/DriveControl_LineAndArc/DriveControlProSVC_impl.cpp.org	  2011-06-28 21:13:58.896590326 +0900
--- RTC/DriveControl_LineAndArc/DriveControlProSVC_impl.cpp		  2011-06-28 21:14:01.446501472 +0900
@@ -13,6 +13,7 @@
  */
 
 #include "DriveControlProSVC_impl.h"
+#include <stdio.h>
 
 /*
  * Example implementational code for IDL interface Drive
--- RTC/DriveControl_Spline/DriveControlProSVC_impl.cpp.org   2011-06-28 21:20:34.462800519 +0900
+++ RTC/DriveControl_Spline/DriveControlProSVC_impl.cpp	      2011-06-28 21:20:50.342246721 +0900
@@ -13,6 +13,7 @@
  */
 
 #include "DriveControlProSVC_impl.h"
+#include <stdio.h>
 
 /*
  * Example implementational code for IDL interface Drive
--- RTC/MapBuilder/Makefile.org	     2011-06-28 21:24:18.695016742 +0900
+++ RTC/MapBuilder/Makefile		     2011-06-28 21:24:55.063755843 +0900
@@ -32,7 +32,7 @@
 TOP=$(OPENHRPHOME)
 #
 #OpenHRP3.1 add
-CORBAOBJ=$(TOP)/hrplib/hrpCorba/CMakeFiles/hrpCorbaStubSkel-3.1.dir
+CORBAOBJ=$(OPENHRPHOME)/build/openhrp-aist-grx/hrplib/hrpCorba/CMakeFiles/hrpCorbaStubSkel-3.1.dir
 ##### 
 
 ##### modify(OpenHRP3.1 Library) ###########################
@@ -38,7 +38,7 @@
 ##### modify(OpenHRP3.1 Library) ###########################
 #CXXFLAGS = `rtm-config --cflags` -I.
 #LDFLAGS  = `rtm-config --libs`
-CXXFLAGS = `rtm-config --cflags` -I/usr/local/include/OpenHRP-3.1/hrpCorba/
+CXXFLAGS = `rtm-config --cflags` -I$(OPENHRPHOME)/include/OpenHRP-3.1/hrpCorba
 LDFLAGS  = `rtm-config --libs` $(CORBAOBJ)/CollisionDetectorSk.o  $(CORBAOBJ)/OpenHRPCommonSk.o   -lomniORB4 -lomnithread $(CORBAOBJ)/ModelLoaderSk.o  $(CORBAOBJ)/DynamicsSimulatorSk.o $(CORBAOBJ)/OnlineViewerSk.o $(CORBAOBJ)/ControllerSk.o $(CORBAOBJ)/ViewSimulatorSk.o $(CORBAOBJ)/WorldSk.o $(CORBAOBJ)/ClockGeneratorSk.o
 ##### 
 SHFLAGS  = -shared
