--- src/lib/rtm/idl/SDOPackage.idl.orig	2011-06-24 04:27:28.256759842 +0900
+++ src/lib/rtm/idl/SDOPackage.idl	2012-02-14 03:06:06.278071680 +0900
@@ -36,9 +36,6 @@
 #ifdef USE_MONITORING
 #include <CosNotifyComm.idl>
 #endif
-#ifndef ORBIT2_IDL
-#define TypeCode CORBA::TypeCode
-#endif
 /** CORBA specific model for SDOs */
 
 #pragma prefix "org.omg"
@@ -121,7 +118,7 @@
   {
     string         name;
 #ifndef ORBIT2_IDL
-    TypeCode       type;
+    CORBA::TypeCode       type;
 #else
     TypeCode       type;
 #endif
