--- src/lib/rtm/idl/SDOPackage.idl.org	2011-07-25 17:01:15.048004888 +0900
+++ src/lib/rtm/idl/SDOPackage.idl	2011-07-25 17:07:42.429088539 +0900
@@ -36,9 +36,6 @@
 #ifdef USE_MONITORING
 #include <CosNotifyComm.idl>
 #endif
-#ifndef ORBIT2_IDL
-#define TypeCode CORBA::TypeCode
-#endif
 /** CORBA specific model for SDOs */
 
 #pragma prefix "org.omg"
@@ -120,7 +117,7 @@
   struct Parameter
   {
     string         name;
-    TypeCode       type;
+    CORBA::TypeCode       type;
     AllowedValues  allowed_values;
   };
 
