--- OpenRTM_aist/SystemLogger.py.org	2012-02-20 12:08:47.227641178 +0900
+++ OpenRTM_aist/SystemLogger.py	2012-02-20 12:08:33.499640785 +0900
@@ -142,16 +142,16 @@
     logging.addLevelName(logging.TRACE,     "TRACE")
     logging.addLevelName(logging.FATAL,     "FATAL")
 
-    logging.root.setLevel([logging.NOTSET,
-                           logging.PARANOID,
-                           logging.VERBOSE,
-                           logging.TRACE,
-                           logging.DEBUG,
-                           logging.INFO,
-                           logging.WARNING,
-                           logging.ERROR,
-                           logging.FATAL,
-                           logging.CRITICAL])
+    # logging.root.setLevel([logging.NOTSET,
+    #                        logging.PARANOID,
+    #                        logging.VERBOSE,
+    #                        logging.TRACE,
+    #                        logging.DEBUG,
+    #                        logging.INFO,
+    #                        logging.WARNING,
+    #                        logging.ERROR,
+    #                        logging.FATAL,
+    #                        logging.CRITICAL])
 
     
     formatter = logging.Formatter('%(asctime)s %(name)s %(levelname)s %(message)s')
