diff -crN example/component/SendImage/Makefile example.editted/component/SendImage/Makefile
*** example/component/SendImage/Makefile	2011-09-09 14:00:23.000000000 +0900
--- example.editted/component/SendImage/Makefile	2012-01-19 22:12:39.433760430 +0900
***************
*** 17,28 ****
  #
  # $Id$
  #
! OPENCVINCDIR = /usr/include/opencv
  VGRINCDIR = ../../../src/lib/libopenvgr
  
  OPENCVLIBS = `pkg-config opencv --libs`
  
! CXXFLAGS = -Wall -g `rtm-config --cflags` -I. -I$(VGRINCDIR) -I$(OPENCVINCDIR)
  LDFLAGS  = `rtm-config --libs` $(OPENCVLIBS)
  SHFLAGS  = -shared
  
--- 17,28 ----
  #
  # $Id$
  #
! OPENCVINCDIR = `pkg-config --cflags opencv`
  VGRINCDIR = ../../../src/lib/libopenvgr
  
  OPENCVLIBS = `pkg-config opencv --libs`
  
! CXXFLAGS = -Wall -g `rtm-config --cflags` -I. -I$(VGRINCDIR) $(OPENCVINCDIR)
  LDFLAGS  = `rtm-config --libs` $(OPENCVLIBS)
  SHFLAGS  = -shared
  
diff -crN example/component/SendImage/Makefile.SendImage example.editted/component/SendImage/Makefile.SendImage
*** example/component/SendImage/Makefile.SendImage	2011-09-09 14:00:23.000000000 +0900
--- example.editted/component/SendImage/Makefile.SendImage	2012-01-19 22:12:39.433760430 +0900
***************
*** 17,28 ****
  #
  # $Id$
  #
! OPENCVINCDIR = /usr/include/opencv
  VGRINCDIR = ../../../src/lib/libopenvgr
  
  OPENCVLIBS = `pkg-config opencv --libs`
  
! CXXFLAGS = -Wall -g `rtm-config --cflags` -I. -I$(VGRINCDIR) -I$(OPENCVINCDIR)
  LDFLAGS  = `rtm-config --libs` $(OPENCVLIBS)
  SHFLAGS  = -shared
  
--- 17,28 ----
  #
  # $Id$
  #
! OPENCVINCDIR = `pkg-config --cflags opencv`
  VGRINCDIR = ../../../src/lib/libopenvgr
  
  OPENCVLIBS = `pkg-config opencv --libs`
  
! CXXFLAGS = -Wall -g `rtm-config --cflags` -I. -I$(VGRINCDIR) $(OPENCVINCDIR)
  LDFLAGS  = `rtm-config --libs` $(OPENCVLIBS)
  SHFLAGS  = -shared
  
