*** src.raw/lib/libopenvgr/Makefile	2012-09-24 11:27:39.179440943 +0900
--- src/lib/libopenvgr/Makefile	2012-09-24 11:30:08.227435669 +0900
***************
*** 10,17 ****
  .cpp.o:
  	$(CXX) -c $(CPPFLAGS) $(CXXFLAGS) $(INCLUDES) $(DEFS) -o $@ $<
  
! OPENCVINC = -I/usr/include/opencv
! RTMINC = -I`rtm-config --prefix`/include/rtm/idl
  INCLUDES = $(OPENCVINC) $(RTMINC)
  
  CC = gcc
--- 10,19 ----
  .cpp.o:
  	$(CXX) -c $(CPPFLAGS) $(CXXFLAGS) $(INCLUDES) $(DEFS) -o $@ $<
  
! # OPENCVINC = -I/usr/include/opencv
! # RTMINC = -I`rtm-config --prefix`/include/rtm/idl
! OPENCVINC = `pkg-config --cflags opencv`
! RTMINC = -I`rtm-config --rtm-includedir`/rtm/idl
  INCLUDES = $(OPENCVINC) $(RTMINC)
  
  CC = gcc
