Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt	(revision 12)
+++ CMakeLists.txt	(working copy)
@@ -54,6 +59,10 @@
 
 include_directories(${PROJECT_SOURCE_DIR}/hrplib)
 
+if(ENABLE_INSTALL_RPATH)
+  set(CMAKE_INSTALL_RPATH "$ORIGIN/../lib:$ORIGIN/../../../lib:${OPENHRP_LIBRARY_DIRS}")
+endif()
+
 set(JAVAC javac)
 set(JAR jar)
 set(JAR jar)
